oci/archive/reader.go

```diff
@@ -0,0 +1,124 @@
+package archive
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/containers/image/v5/internal/tmpdir"
+	"github.com/containers/image/v5/oci/internal"
+	"github.com/containers/image/v5/oci/layout"
+	"github.com/containers/image/v5/transports"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/pkg/archive"
+	imgspecv1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+// Reader manages the temp directory that the oci archive is untarred to and the
+// manifest of the images. It allows listing its contents and accessing
+// individual images with less overhead than creating image references individually
+// (because the archive is, if necessary, copied or decompressed only once)
+type Reader struct {
+	manifest      *imgspecv1.Index
+	tempDirectory string
+	path          string // The original, user-specified path
+}
+
+// NewReader returns a Reader for src. The caller should call Close() on the returned object
+func NewReader(ctx context.Context, sys *types.SystemContext, ref types.ImageReference) (*Reader, error) {
+	standalone, ok := ref.(ociArchiveReference)
+	if !ok {
+		return nil, errors.Errorf("Internal error: NewReader called for a non-oci/archive ImageReference %s", transports.ImageName(ref))
```

Comment:
`fmt.Errorf` in new code, please. (In all instances)

Reply:
fixed